aten/src/ATen/native/BinaryOps.cpp

```diff
@@ -203,5 +204,48 @@ Tensor& logical_xor_(Tensor& self, const Tensor& other) {
   return native::logical_xor_out(self, self, other);
 }
 
+Tensor& lt_out(Tensor& result, const Tensor& self, const Tensor& other) {
+  TORCH_CHECK(result.scalar_type() == kBool,
```

Comment:
It might be better to do this in `TensorIterator::comparison_op`. Also, it would be good to tell the user what the dtype of the tensor that they passed in was.

Reply:
We should avoid this check when do inplace operation.

Reply:
Oh, I see. This is fine for now but we should aim to deduplicate code between all the comparison op implementations.